### PR TITLE
Api spec for Visa validation

### PR DESCRIPTION
**Pull request by dara123**

### CODE REVIEW

File: spec/definitions/PaymentCardVerification.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  AVSResult:
+    description: Address verification result
+    type: string
+    readOnly: true
+  CVVResult:
+    description: CVV result
+    type: string
+    readOnly: true
+  billingAddress:
+    description: The Billing Address. Required if card is creating not from Token
```
Comment (colinbird) [inline spec/definitions/PaymentCardVerification.yaml]: "Required if card is not created from Token"

Comment (dara123) [reply spec/definitions/PaymentCardVerification.yaml]: removed

File: spec/definitions/PaymentCardVerification.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+type: object
+properties:
+  id:
+    description: Verfication payment card id
```
Comment (adamaltman) [inline spec/definitions/PaymentCardVerification.yaml]: typo

File: spec/paths/payment-cards@{id}@verification.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+parameters:
+  - $ref: "#/parameters/resourceId"
+post:
+  tags:
+    - Payment Cards
+  summary: Verify a Payment Card
+  description: |
+    Verfiy a Payment Card
```
Comment (adamaltman) [inline spec/paths/payment-cards@{id}@verification.yaml]: typo.  Visa also uses the term validate.  I don't know what the difference is...

File: spec/paths/payment-cards@validation.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+get:
```
Comment (adamaltman) [inline spec/paths/payment-cards@validation.yaml]: Here you change the name and the path.  I don't understand it.

**Comment by colinbird** [comment]
We can use Validation instead of Verification

File: spec/definitions/PaymentInstrumentValidation.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+properties:
+  id:
+    description: Validation payment instrument
+    readOnly: true
+    allOf:
+      - $ref: "#/definitions/ResourceId"
+  method:
+    description: Payment method
+    type: string
+  paymentInstrumentId:
```
Comment (adamaltman) [inline spec/definitions/PaymentInstrumentValidation.yaml]: is this required?

File: spec/definitions/PaymentInstrumentValidation.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
+type: object
+properties:
+  id:
+    description: Validation payment instrument
+    readOnly: true
+    allOf:
+      - $ref: "#/definitions/ResourceId"
+  method:
```
Comment (adamaltman) [inline spec/definitions/PaymentInstrumentValidation.yaml]: is this required?

File: spec/paths/payment-instrument-validation@{id}.yaml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        Rate-Limit-Reset:
+          $ref: "#/headers/Rate-Limit-Reset"
+        Pagination-Total:
+          $ref: "#/headers/Pagination-Total"
+        Pagination-Limit:
+          $ref: "#/headers/Pagination-Limit"
+        Pagination-Offset:
+          $ref: "#/headers/Pagination-Offset"
+      schema:
+        type: array
```
Comment (adamaltman) [inline spec/paths/payment-instrument-validation@{id}.yaml]: I think you have this backwards.  The array should be returned for the `get request to payment-instrument-validation/`.  This should return a single resource.

File: spec/definitions/PaymentInstrumentValidation.yaml (hunk 5 above)
Comment (dara123) [reply spec/definitions/PaymentInstrumentValidation.yaml]: Fixed it in POST. Should we add required in definition too?

File: spec/paths/payment-instrument-validation.yaml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    - name: body
+      in: body
+      description: payment instrument
+      schema:
+        type: object
+        properties:
+          method:
+            description: Payment method
+            type: string
+            required: true
```
Comment (adamaltman) [inline spec/paths/payment-instrument-validation.yaml]: This is not OpenAPI.  Not sure what this is...

**Pull request merged**